Sources/SwiftSyntax/SyntaxProtocol.swift

```diff
@@ -241,14 +237,18 @@ extension SyntaxProtocol {
 
   /// Returns this node or the first ancestor that satisfies `condition`.
   public func ancestorOrSelf<T>(mapping map: (Syntax) -> T?) -> T? {
-    var walk: Syntax? = Syntax(self)
-    while let unwrappedParent = walk {
-      if let mapped = map(unwrappedParent) {
-        return mapped
-      }
-      walk = unwrappedParent.parent
+    return withUnownedSyntax(self) {
+      var node = $0
+      repeat {
```

Comment:
If this is an infinite loop, would `while true` be easier to read than `repeat { } while true`?